.travis.yml

```diff
@@ -1,38 +1,124 @@
-language: c
 sudo: false
 
-git:
-  submodules: false  # whether to recursively clone submodules
+language: generic
 
 cache:
   directories:
-    - $HOME/.cabal/packages
-    - $HOME/.mafia
-
-before_cache:
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/build-reports.log
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/00-index.*
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/*.json
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/01-index.cache
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/01-index.tar
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/01-index.tar.idx
+  - $HOME/.ghc
+  - $HOME/.cabal
+  - $HOME/.stack
 
 matrix:
   include:
-    - compiler: "ghc-8.0.1"
-      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-1.24,ghc-8.0.1], sources: [hvr-ghc]}}
-    - compiler: "ghc-8.0.2"
-      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-1.24,ghc-8.0.2], sources: [hvr-ghc]}}
-    - compiler: "ghc-8.2.1"
-      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-1.24,ghc-8.0.2], sources: [hvr-ghc]}}
+  # The Cabal builds.
+  - env: BUILD=cabal GHCVER=8.0.2 CABALVER=1.24 HAPPYVER=1.19.5 ALEXVER=3.1.7
+    compiler: ": #GHC 8.0.2"
```

Comment:
Could we also get a 8.2.1 build?

Reply:
Sure thing, I'll just need to work out how

Reply:
Should we just do 8.2.1 instead of 8.0.2? Or both in tandem?

Reply:
Looks like just copying this entry and replacing the strings `8.0.2` -> `8.2.1` will work!

Reply:
Nice! How did you test that so quickly?

Reply:
Would we need Cabal 2 for `8.2.1` or is `1.24` fine?

Reply:
I'm currently using 1.24 with 8.2.1 without any problems.

Reply:
Looks like it built fine; just waiting on `stack default` for OS X  now.